### ISSUE TITLE
2x2 complex exponential of zeros matrix is undefined

### ISSUE THREAD

**Issue by sdaxen** [JuliaArrays/StaticArrays.jl#623]
Exponential of sized complex 2x2 matrix of zeros produces `NaN`s. This is not the case for a regular `Matrix`.

```julia
julia> using LinearAlgebra, StaticArrays

julia> exp(@SMatrix zeros(ComplexF64, 2, 2))
2×2 SArray{Tuple{2,2},Complex{Float64},2,4}:
 NaN+NaN*im  NaN+NaN*im
 NaN+NaN*im  NaN+NaN*im

julia> exp(@SMatrix zeros(ComplexF64, 1, 1))
1×1 SArray{Tuple{1,1},Complex{Float64},2,1}:
 1.0 + 0.0im

julia> exp(@SMatrix zeros(ComplexF64, 3, 3))
3×3 SArray{Tuple{3,3},Complex{Float64},2,9}:
 1.0+0.0im  0.0+0.0im  0.0+0.0im
 0.0+0.0im  1.0+0.0im  0.0+0.0im
 0.0+0.0im  0.0+0.0im  1.0+0.0im

julia> exp(zeros(ComplexF64, 2, 2))
2×2 Array{Complex{Float64},2}:
 1.0+0.0im  0.0+0.0im
 0.0-0.0im  1.0+0.0im

julia> exp(@SMatrix zeros(Float64, 2, 2))
2×2 SArray{Tuple{2,2},Float64,2,4}:
 1.0  0.0
 0.0  1.0
```

**Comment by mschauer** [comment]
`(a-d)^2 + 4*b*c == 0` should be special cased like in the real case, to avoid division by zero in

https://github.com/JuliaArrays/StaticArrays.jl/blob/86c16e323c3b9ed6c7812f0e900368a5c918ccde/src/expm.jl#L55

**Comment by sdaxen** [comment]
Just submitted a pull request with special-casing.

**Comment by mschauer** [comment]
Closed by #624
